cmd/artifact/command/push.go

```diff
@@ -24,12 +25,26 @@ func pushCommand(options *Options) *cobra.Command {
 			var err error
 			ctx := context.Background()
 
-			if releaseManagerClient.Metadata.AuthToken != "" {
-				artifactID, err = flow.PushArtifactToReleaseManager(ctx, &releaseManagerClient, options.FileName, options.RootPath)
-				if err != nil {
-					return err
-				}
+			idpURL := os.Getenv("HAMCTL_OAUTH_IDP_URL")
```

Comment:
Per conversation: this should be flags and hamctl prefix should be removed

Reply:
Thinking about it, I don't like to give the client secret as a command line argument as that leaks out to e.g. /proc/pid/cmdline. I will put the secret in a file and take an argument with a path to that file 🚀

Reply:
As all secrets are currently mounted as args I would argue we should follow that convention for now.

We can then iterate on a follow up changing all of them in one go.